main

```diff
@@ -262,11 +304,39 @@ run_tests() {
 
     local rc=0
     local timeout=$(cat state/parsed/timeout)
-    timeout --kill-after=30s $timeout \
-        ssh -q -o StrictHostKeyChecking=no \
-               -o UserKnownHostsFile=/dev/null \
-               root@$node_addr "sh worker.sh 2>&1" | \
-        tee -a $upload_dir/output.txt || rc=$?
+
+    if ! containerized; then
+        timeout --kill-after=30s "$timeout" \
+            ssh -q -o StrictHostKeyChecking=no \
+                   -o UserKnownHostsFile=/dev/null \
+                   root@$node_addr "sh worker.sh 2>&1" | \
+            tee -a $upload_dir/output.txt || rc=$?
+    else
+        local mnt=$PWD/state/cnt
+
+        # We use below to make it more convenient for running unprivileged on
+        # dev machines. Though because we use timeout, sudo doesn't have control
+        # of the TTY, so we use it beforehand so it can cache credentials.
+        sudo true
+
+        # Setting a timeout on docker run is not reliable since it's the daemon
+        # running it. And we don't want to trust the timeout *inside* the
+        # container as well. So we follow up with a docker kill.
+        timeout --kill-after=30s "$timeout" \
+            sudo docker run --rm \
+                --workdir / \
+                --cidfile state/cid \
+                -v $mnt/checkout:/checkout:z \
+                -v $mnt/worker.sh:/worker.sh:z \
+                -v $mnt/rhci-extras.repo:/etc/yum.repos.d/rhci-extras.repo:z \
+                "$(cat state/parsed/image)" sh -c "sh worker.sh 2>&1" | \
```

Comment:
Right, I did do a scan of all outside inputs to make sure they were handled & quoted properly when passed around. Though I will likely implement some sanity checking in the yml parser itself.